crates/mipsy_parser/src/misc.rs

```diff
@@ -31,21 +31,24 @@ where
             if leftover.is_empty() {
                 Ok(t)
             } else {
-                match comment_multispace0(leftover) {
-                    Ok((leftover, _)) => {
-                        Err(ErrorLocation {
-                            file_name,
-                            line: leftover.location_line(),
-                            col:  leftover.get_column(),
-                        })
-                    }
-                    Err(err) => {
-                        eprintln!("ERROR: {}", err);
-                        panic!("this should never happen - please report an issue at https://github.com/insou22/mipsy")            
-                    }
-                }
+                parse_multispace(leftover, file_name)
             }
         }
+        Err(_) => {
+            parse_multispace(i, file_name)
+        }
+    }
+}
+
+fn parse_multispace<T>(i: nom_locate::LocatedSpan<&[u8]>, file_name: Option<Rc<str>>) -> Result<T, ErrorLocation> {
```

Comment:
maybe `fn leftover_tokens_strip_multispace` or something?
also, no need to type `nom_locate::LocatedSpan<&[u8]>`, we have the type alias `Span<'a>`, you'll just have to make the `fn` generic over `'a` (which is already what is happening with lifetime elision) -- so something like `fn leftover_tokens_strip_multispace<'a>(i: Span<'a>, ...) ...`

It also is a bit confusing to return a `forall T. Result<T, ErrorLocation>`, given it only returns the `Err` variant -- this is only compiling due to the compiler being clever, similar to never type (`!`) promotion. I would just return the `ErrorLocation`, and then wrap that in `Err(...)` at the call sites in the `parse_result` function